ESLint error in my React Native (javascript) project

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Whenever I save / change my zed settings (~/.config/zed) I get this ESLint error, I'm not even sure how to disable the linter or attempt a fix at this issue myself.

ESLint: prettier.resolveConfig.sync is not a function Occurred while linting /Users/macbook/Desktop/gps/gps/index.js:5 Rule: "prettier/prettier". Please see the 'ESLint' output channel for details.


### Environment

Zed: v0.123.3 (Zed)
OS: macOS 14.3.1
Memory: 8 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature


<img width="279" alt="image" src="https://github.com/zed-industries/zed/assets/31011142/cde59033-a880-405b-813b-6184f7dad352">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.


2024-02-23T00:51:22+02:00 [INFO] Initializing default prettier with plugins {}
2024-02-23T00:51:22+02:00 [INFO] Initializing default prettier with plugins {"prettier-plugin-tailwindcss"}
2024-02-23T00:51:22+02:00 [INFO] 13 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/App.tsx",
  "state": 3
}
2024-02-23T00:51:22+02:00 [INFO] 13 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/app/screens/Login.js",
  "state": 3
}
2024-02-23T00:51:23+02:00 [INFO] 13 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/index.js",
  "state": 3
}
2024-02-23T00:51:31+02:00 [INFO] reload git repository ".git"
2024-02-23T00:51:31+02:00 [INFO] reload git repository ".git"
2024-02-23T00:52:31+02:00 [INFO] reload git repository ".git"
2024-02-23T00:52:31+02:00 [INFO] reload git repository ".git"
2024-02-23T00:52:58+02:00 [INFO] 13 unhandled notification eslint/showOutputChannel:
[
  null
]
2024-02-23T00:52:59+02:00 [INFO] 13 unhandled notification eslint/showOutputChannel:
[
  null
]
2024-02-23T00:52:59+02:00 [INFO] 13 unhandled notification eslint/showOutputChannel:
[
  null
]
2024-02-23T00:53:14+02:00 [INFO] stopping language server tailwindcss-language-server
2024-02-23T00:53:14+02:00 [INFO] stopping language server typescript-language-server
2024-02-23T00:53:14+02:00 [INFO] stopping language server eslint
2024-02-23T00:53:14+02:00 [INFO] Initializing default prettier with plugins {}
2024-02-23T00:53:14+02:00 [INFO] Initializing default prettier with plugins {"prettier-plugin-tailwindcss"}
2024-02-23T00:53:14+02:00 [INFO] starting language server "typescript-language-server", path: "/Users/macbook/Desktop/gps/gps", id: 15
2024-02-23T00:53:14+02:00 [INFO] starting language server "eslint", path: "/Users/macbook/Desktop/gps/gps", id: 16
2024-02-23T00:53:14+02:00 [INFO] starting language server "tailwindcss-language-server", path: "/Users/macbook/Desktop/gps/gps", id: 17
2024-02-23T00:53:14+02:00 [ERROR] crates/lsp/src/lsp.rs:648: oneshot canceled
2024-02-23T00:53:14+02:00 [ERROR] crates/lsp/src/lsp.rs:648: oneshot canceled
2024-02-23T00:53:14+02:00 [ERROR] crates/lsp/src/lsp.rs:648: oneshot canceled
2024-02-23T00:53:15+02:00 [INFO] 16 unhandled notification window/logMessage:
{
  "type": 3,
  "message": "ESLint server running in node v18.15.0"
}
2024-02-23T00:53:15+02:00 [INFO] 15 unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Using Typescript version (user-setting) 5.2.2 from path \"/Users/macbook/Desktop/gps/gps/node_modules/typescript/lib/tsserver.js\""
}
2024-02-23T00:53:15+02:00 [INFO] 15 unhandled notification $/typescriptVersion:
{
  "version": "5.2.2",
  "source": "user-setting"
}
2024-02-23T00:53:21+02:00 [INFO] 16 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/App.tsx",
  "state": 3
}
2024-02-23T00:53:22+02:00 [INFO] 16 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/app/screens/Login.js",
  "state": 3
}
2024-02-23T00:53:22+02:00 [INFO] 16 unhandled notification eslint/status:
{
  "uri": "file:///Users/macbook/Desktop/gps/gps/index.js",
  "state": 3
}

--- a/crates/gpui_macros/src/style_helpers.rs
+++ b/crates/gpui_macros/src/style_helpers.rs
@@ -399,6 +399,8 @@ fn box_suffixes() -> Vec<(&'static str, TokenStream2, &'static str)> {
         ("72", quote! { rems(18.) }, "288px (18rem)"),
         ("80", quote! { rems(20.) }, "320px (20rem)"),
         ("96", quote! { rems(24.) }, "384px (24rem)"),
+        ("112", quote! { rems(28.) }, "448px (28rem)"),
+        ("128", quote! { rems(32.) }, "512px (32rem)"),
         ("auto", quote! { auto() }, "Auto"),
         ("px", quote! { px(1.) }, "1px"),
         ("full", quote! { relative(1.) }, "100%"),

--- a/crates/project/src/project.rs
+++ b/crates/project/src/project.rs
@@ -229,6 +229,7 @@ pub struct LanguageServerPromptRequest {
     pub level: PromptLevel,
     pub message: String,
     pub actions: Vec<MessageActionItem>,
+    pub lsp_name: String,
     response_channel: Sender<MessageActionItem>,
 }
 
@@ -3022,6 +3023,7 @@ impl Project {
             cx.update(|cx| adapter.workspace_configuration(worktree_path, cx))?;
         let language_server = pending_server.task.await?;
 
+        let name = language_server.name();
         language_server
             .on_notification::<lsp::notification::PublishDiagnostics, _>({
                 let adapter = adapter.clone();
@@ -3160,8 +3162,10 @@ impl Project {
         language_server
             .on_request::<lsp::request::ShowMessageRequest, _, _>({
                 let this = this.clone();
+                let name = name.to_string();
                 move |params, mut cx| {
                     let this = this.clone();
+                    let name = name.to_string();
                     async move {
                         if let Some(actions) = params.actions {
                             let (tx, mut rx) = smol::channel::bounded(1);
@@ -3174,6 +3178,7 @@ impl Project {
                                 message: params.message,
                                 actions,
                                 response_channel: tx,
+                                lsp_name: name.clone(),
                             };
 
                             if let Ok(_) = this.update(&mut cx, |_, cx| {
@@ -3211,6 +3216,7 @@ impl Project {
                 }
             })
             .detach();
+
         let mut initialization_options = adapter.adapter.initialization_options();
         match (&mut initialization_options, override_options) {
             (Some(initialization_options), Some(override_options)) => {

--- a/crates/ui/src/components/label/label_like.rs
+++ b/crates/ui/src/components/label/label_like.rs
@@ -7,6 +7,7 @@ use crate::prelude::*;
 pub enum LabelSize {
     #[default]
     Default,
+    Large,
     Small,
     XSmall,
 }
@@ -97,6 +98,7 @@ impl RenderOnce for LabelLike {
                 )
             })
             .map(|this| match self.size {
+                LabelSize::Large => this.text_ui_lg(),
                 LabelSize::Default => this.text_ui(),
                 LabelSize::Small => this.text_ui_sm(),
                 LabelSize::XSmall => this.text_ui_xs(),

--- a/crates/ui/src/styled_ext.rs
+++ b/crates/ui/src/styled_ext.rs
@@ -35,6 +35,17 @@ pub trait StyledExt: Styled + Sized {
         self.text_size(size.rems())
     }
 
+    /// The large size for UI text.
+    ///
+    /// `1rem` or `16px` at the default scale of `1rem` = `16px`.
+    ///
+    /// Note: The absolute size of this text will change based on a user's `ui_scale` setting.
+    ///
+    /// Use `text_ui` for regular-sized text.
+    fn text_ui_lg(self) -> Self {
+        self.text_size(UiTextSize::Large.rems())
+    }
+
     /// The default size for UI text.
     ///
     /// `0.825rem` or `14px` at the default scale of `1rem` = `16px`.

--- a/crates/ui/src/styles/typography.rs
+++ b/crates/ui/src/styles/typography.rs
@@ -13,6 +13,13 @@ pub enum UiTextSize {
     /// Note: The absolute size of this text will change based on a user's `ui_scale` setting.
     #[default]
     Default,
+    /// The large size for UI text.
+    ///
+    /// `1rem` or `16px` at the default scale of `1rem` = `16px`.
+    ///
+    /// Note: The absolute size of this text will change based on a user's `ui_scale` setting.
+    Large,
+
     /// The small size for UI text.
     ///
     /// `0.75rem` or `12px` at the default scale of `1rem` = `16px`.
@@ -31,6 +38,7 @@ pub enum UiTextSize {
 impl UiTextSize {
     pub fn rems(self) -> Rems {
         match self {
+            Self::Large => rems(16. / 16.),
             Self::Default => rems(14. / 16.),
             Self::Small => rems(12. / 16.),
             Self::XSmall => rems(10. / 16.),

--- a/crates/workspace/src/notifications.rs
+++ b/crates/workspace/src/notifications.rs
@@ -1,10 +1,14 @@
 use crate::{Toast, Workspace};
 use collections::HashMap;
 use gpui::{
-    AnyView, AppContext, AsyncWindowContext, DismissEvent, Entity, EntityId, EventEmitter, Global,
-    PromptLevel, Render, Task, View, ViewContext, VisualContext, WindowContext,
+    svg, AnyView, AppContext, AsyncWindowContext, DismissEvent, Entity, EntityId, EventEmitter,
+    Global, PromptLevel, Render, Task, View, ViewContext, VisualContext, WindowContext,
 };
+use language::DiagnosticSeverity;
+
 use std::{any::TypeId, ops::DerefMut};
+use ui::prelude::*;
+use util::ResultExt;
 
 pub fn init(cx: &mut AppContext) {
     cx.set_global(NotificationTracker::new());
@@ -167,6 +171,105 @@ impl Workspace {
             });
     }
 }
+
+pub struct LanguageServerPrompt {
+    request: Option<project::LanguageServerPromptRequest>,
+}
+
+impl LanguageServerPrompt {
+    pub fn new(request: project::LanguageServerPromptRequest) -> Self {
+        Self {
+            request: Some(request),
+        }
+    }
+
+    async fn select_option(this: View<Self>, ix: usize, mut cx: AsyncWindowContext) {
+        util::async_maybe!({
+            let potential_future = this.update(&mut cx, |this, _| {
+                this.request.take().map(|request| request.respond(ix))
+            });
+
+            potential_future? // App Closed
+                .ok_or_else(|| anyhow::anyhow!("Response already sent"))?
+                .await
+                .ok_or_else(|| anyhow::anyhow!("Stream already closed"))?;
+
+            this.update(&mut cx, |_, cx| cx.emit(DismissEvent))?;
+
+            anyhow::Ok(())
+        })
+        .await
+        .log_err();
+    }
+}
+
+impl Render for LanguageServerPrompt {
+    fn render(&mut self, cx: &mut ViewContext<Self>) -> impl IntoElement {
+        let Some(request) = &self.request else {
+            return div().id("language_server_prompt_notification");
+        };
+
+        h_flex()
+            .id("language_server_prompt_notification")
+            .elevation_3(cx)
+            .items_start()
+            .p_2()
+            .gap_2()
+            .w_full()
+            .child(
+                v_flex()
+                    .overflow_hidden()
+                    .child(
+                        h_flex()
+                            .children(
+                                match request.level {
+                                    PromptLevel::Info => None,
+                                    PromptLevel::Warning => Some(DiagnosticSeverity::WARNING),
+                                    PromptLevel::Critical => Some(DiagnosticSeverity::ERROR),
+                                }
+                                .map(|severity| {
+                                    svg()
+                                        .size(cx.text_style().font_size)
+                                        .flex_none()
+                                        .mr_1()
+                                        .map(|icon| {
+                                            if severity == DiagnosticSeverity::ERROR {
+                                                icon.path(IconName::ExclamationTriangle.path())
+                                                    .text_color(Color::Error.color(cx))
+                                            } else {
+                                                icon.path(IconName::ExclamationTriangle.path())
+                                                    .text_color(Color::Warning.color(cx))
+                                            }
+                                        })
+                                }),
+                            )
+                            .child(
+                                Label::new(format!("{}:", request.lsp_name))
+                                    .size(LabelSize::Default),
+                            ),
+                    )
+                    .child(Label::new(request.message.to_string()))
+                    .children(request.actions.iter().enumerate().map(|(ix, action)| {
+                        let this_handle = cx.view().clone();
+                        ui::Button::new(ix, action.title.clone())
+                            .size(ButtonSize::Large)
+                            .on_click(move |_, cx| {
+                                let this_handle = this_handle.clone();
+                                cx.spawn(|cx| async move {
+                                    LanguageServerPrompt::select_option(this_handle, ix, cx).await
+                                })
+                                .detach()
+                            })
+                    })),
+            )
+            .child(
+                ui::IconButton::new("close", ui::IconName::Close)
+                    .on_click(cx.listener(|_, _, cx| cx.emit(gpui::DismissEvent))),
+            )
+    }
+}
+
+impl EventEmitter<DismissEvent> for LanguageServerPrompt {}
 
 pub mod simple_message_notification {
     use gpui::{

--- a/crates/workspace/src/workspace.rs
+++ b/crates/workspace/src/workspace.rs
@@ -59,7 +59,10 @@ use std::{
     any::TypeId,
     borrow::Cow,
     cell::RefCell,
-    cmp, env,
+    cmp,
+    collections::hash_map::DefaultHasher,
+    env,
+    hash::{Hash, Hasher},
     path::{Path, PathBuf},
     rc::Rc,
     sync::{atomic::AtomicUsize, Arc, Weak},
@@ -579,24 +582,13 @@ impl Workspace {
                 }),
 
                 project::Event::LanguageServerPrompt(request) => {
-                    let request = request.clone();
+                    let mut hasher = DefaultHasher::new();
+                    request.message.as_str().hash(&mut hasher);
+                    let id = hasher.finish();
 
-                    cx.spawn(|_, mut cx| async move {
-                        let messages = request
-                            .actions
-                            .iter()
-                            .map(|action| action.title.as_str())
-                            .collect::<Vec<_>>();
-                        let index = cx
-                            .update(|cx| {
-                                cx.prompt(request.level, "", Some(&request.message), &messages)
-                            })?
-                            .await?;
-                        request.respond(index).await;
-
-                        Result::<(), anyhow::Error>::Ok(())
-                    })
-                    .detach()
+                    this.show_notification(id as usize, cx, |cx| {
+                        cx.new_view(|_| notifications::LanguageServerPrompt::new(request.clone()))
+                    });
                 }
 
                 _ => {}
@@ -2766,7 +2758,7 @@ impl Workspace {
                     .z_index(100)
                     .right_3()
                     .bottom_3()
-                    .w_96()
+                    .w_112()
                     .h_full()
                     .flex()
                     .flex_col()

